viewing_party/party.py

```diff
@@ -1,23 +1,158 @@
 # ------------- WAVE 1 --------------------
 
+from imp import new_module
+from turtle import up
+from zoneinfo import available_timezones
+
+
 def create_movie(title, genre, rating):
-    pass
+    """Create a new movie dict with given parameters and return a dict."""
+    new_movie = {}
+    if title == None or genre == None or rating == None:
+        return None
+    new_movie['title'] = title
+    new_movie['genre'] = genre
+    new_movie['rating'] = rating 
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    """Add a movie to a watched list in user data and return user data."""
+    user_data['watched'].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """Add a movie to a whatchlist in user data and return user data."""
+    user_data['watchlist'].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    """Move a movie from a watchlist to watched and return user data."""
+    watchlist = user_data['watchlist']
+    watched = user_data['watched']
+    for film in watchlist:
+        if film['title'] == movie_title:
+            watchlist.remove(film)
+            watched.append(film) 
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    """Find the average rating of all watched movies and return it."""
+    watched = user_data['watched']
+    rating_sum = 0
+
+    for movie in watched:
+        rating_sum  += movie['rating']
+
+    if len(watched) != 0:
+        rating_avg = rating_sum / len(watched)
+        return rating_avg
+    else:
+        return 0.0
+            
+
+def get_most_watched_genre(user_data):
+    """Find the most popular genre among watched movies and return it."""
+    watched = user_data['watched']
+    all_genres_watched = {}
+
+    if watched == []:
+        return None
+    else:

```

Comment:
This `else:` isn't needed. The `if` block at the top is an error check that will end the function, but if the conditional is false, the rest of the function will run.